### DISCUSSION TITLE
Where do I find detailed package documentation?

### DISCUSSION THREAD

**Discussion by chuckwondo** [sile-typesetter/sile#1690] [Q&A] [answered]
I'm totally new to SILE, and I'm only vaguely familiar with TeX, typography, etc.

I'm trying my hand at creating a quarterly publication/booklet using SILE, which is currently produced with InDesign. I have no familiarity with InDesign either, but I'm exploring the possibility of replacing InDesign with SILE.

I'm dabbling around with some of the common packages, but the details in the SILE book regarding the various packages seem to be a bit lacking, and perhaps written for those who may already have some familiarity with things.

For example, the section on the tableofcontents package doesn't seem to provide enough detail for me to actually use it, unless I'm missing something, so I'm wondering if I should be looking at other resources for complete details about the packages.

More specifically, here's basically what I have at this point (abridged):

```
\use[module=packages.pdf]
\use[module=packages.tableofcontents]
\use[module=packages.dropcaps]

\begin[class=book;papersize=6in x 9in]{document}
\tableofcontents

\section[numbering=false]{section heading}
\dropcap{S}ection contents...

\end{document}
```

When I run sile, I get this message:

```
! No localized message for tableofcontents-title found in locale en at ...
```

However, the SILE document doesn't tell me how to specify a title for the TOC, so I just get "nil" as the TOC title.

Where can I find more details about not only the tableofcontents package, but other packages as well? The SILE doc doesn't seem to provide full reference details for the packages it mentions.

**Comment by Omikhleia** [comment, marked as answer]
Greetings,
- As a first thing, the modules/packages should rather be loaded _after_ the document class declaration (that is, in the document)
- You don't need to load the `tableofcontent` package, actually, when using the **book** class: the latter takes care of it by default.
- The localization error is a known bug (#1157); to work-around it, you currently have to set a language explicitly.

Wrapping it all, the following works:
```
\begin[class=book;papersize=6in x 9in]{document}
\use[module=packages.pdf]
\use[module=packages.dropcaps]
\language[main=en]

\tableofcontents

\section[numbering=false]{section heading}
\dropcap{S}ection contents...

\end{document}
```

Regarding the SILE manual, it uses the documentation provided by the packages. Some are pretty good, and others less well documented (there are already a few issues opened on specific packages). The last resort is to read the package code itself (the documentation bit is usually near the end)... Suggesting improvements to the manual and to packages' documentations is, I'd assume, welcome!

EDIT: Forgot that item:
> the SILE document doesn't tell me how to specify a title for the TOC,

The standard tableofcontent package actually provides a localized string (well, when the language is correctly set, due to the above-mentioned issue), which for English defaults to "Table of Contents".

Here is one way to override it to "Contents" (for instance) - from §8.4 "Localization" in the SILE manual v0.14.7:
```
\ftl{tableofcontents-title = Contents}
\tableofcontents
```

**Reply by chuckwondo** [reply]
Great. That did the trick. Thank you!

Regarding reading the package code to figure things out when the documentation is lacking, I'm happy to resort to that, but I don't know how to locate the package code. How do I find it?

**Reply by Omikhleia** [reply]
> I'm happy to resort to that, but I don't know how to locate the package code. How do I find it?

In the `packages` folder in the code: either cloning the repository sources, or finding where your SILE installation has its installed files (which might depend on the system, I dunno... E.g. on my Ubuntu setup it ended in `/usr/share/sile/packages/`). Each package is a folder there, usually containing (at least) an `init.lua` main file, e.g. `dropcaps/init.lua`.

**Reply by chuckwondo** [reply]
Perfect. Thanks.

**Reply by alerque** [reply]
The [package sources](https://github.com/sile-typesetter/sile/tree/master/packages) are also pretty easy to browse here on GitHub if you don't want to muck around on your system.
